salad/steps/browser/alerts.py

```diff
@@ -1,59 +1,127 @@
+# -*- coding: utf-8 -*-
+
 from lettuce import step, world
-from salad.tests.util import assert_with_negate
+
+from salad.logger import logger
+from salad.tests.util import (
+    is_unsupported,
+    assert_with_negate,
+    store_with_case_option,
+    transform_for_upper_lower_comparison,
+    wait_for_completion
+)
 
 
 def _get_alert_or_none():
+    if is_unsupported('phantomjs', 'alerts'):
+        msg = "phantomjs does not support alerts"
+        logger.info(msg)
+        raise NotImplementedError(msg)
+
     try:
-        alert = world.browser.get_alert()
+        alert = world.browser.driver.switch_to_alert()
+        # switch_to_alert() always returns an object, but if there is no
+        # alert, the following line will raise an exception
+        alert.text
```

Comment:
i wanted to explain why i had to add the line "alert.text", although the text is not further used.
get_alert() used to through the exception here, but driver.switch_to_alert() does not
and i also did not want to change the logic that is using the exception